mechanisms/vecevent.mod

```diff
@@ -149,10 +149,10 @@ static void bbcore_read(double* xarray, int* iarray, int* xoffset, int* ioffset,
   ia = iarray + *ioffset;
   dsize = ia[0];
   if (!_p_ptr) {
-    _p_ptr = vector_new1(dsize);
+    _p_ptr = (double*)vector_new(dsize);
```

Comment:
In this case it might be nice to do something like
```c++
IvocVect* pv = (IvocVect*)_p_ptr;
if(!pv) {
  pv = vector_new(dsize);
}
assert(dsize == vector_capacity(pv));
dv = vector_vec(pv);
_p_ptr = (double*)pv;
```
as this makes it clearer that it's only the storage of an `IvocVect*` in `_p_ptr` that is type-unsafe and needs casting, not all of the `vector_*` operations.

Reply:
Done.